src/java.base/share/classes/java/net/Inet6AddressImpl.java

```diff
@@ -148,4 +148,8 @@ public synchronized InetAddress loopbackAddress() {
 
     private InetAddress anyLocalAddress;
     private InetAddress loopbackAddress;
+
+    static {
+        jdk.internal.loader.BootLoader.loadLibrary("net");
+    }
```

Comment:
I have not seen any failures in CI testing. Is there a specific test that would reveal whether this is a problem?

Reply:
It may be because we have no IPv4 only machine in the CI? I seems strange that IPv6 is treated differently than IPv4 in that respect.

Reply:
How would you suggest testing this?

Reply:
I don't know - you added that code to Inet6AddressImpl - so presumably a test was failing without that code?
Which test was that?

Reply:
I'll have to investigate.

Reply:
Possibly - if you made isIPv6Supported() in InetAddress.c return false, you might be able to see the issue in the same test that you observed failing without your change. 

InetAddress has a static block that will load the "net" library, and an other static block that will create the InetAddressImpl. It's a bit curious because I would expect the block that loads the "net" library to be executed before.

And the only place where I see Inet6AddressImpl being used is in that second static block in InetAddress.

I am not an expert on library loading though :-(

Reply:
Thanks for the suggestions. I will look into it.

Reply:
Without loading libnet in Inet6AddressImpl, the test java/net/InetAddress/NullCharInHostnameDriver.java fails with UnsatisfiedLinkError:
```
java.lang.UnsatisfiedLinkError: 'java.net.InetAddress[] java.net.Inet6AddressImpl.lookupAllHostAddr(java.lang.String, int)'
	at java.base/java.net.Inet6AddressImpl.lookupAllHostAddr(Native Method)
	at java.base/java.net.Inet6AddressImpl.lookupAllHostAddr(Inet6AddressImpl.java:52)
	at java.base/java.net.NullCharInHostname.main(NullCharInHostname.java:37)
```

Reply:
OK, this test uses a private API to create an instance of Inet6AddressImpl; This explain why in this test Inet6AddressImpl gets loaded before InetAddress.
```
            var impl = new Inet6AddressImpl();
```
It should never happen outside of this test. Now the tricky question: what in your proposed changes caused "net" not to be loaded before the test created new Inet6AddressImpl(); ?

Reply:
Loading "net" was removed from IOUtil so I am thinking that IOUtil must have been initialized somewhere before constructing Inet6AddressImpl, but I've not identified where just yet.

Reply:
I see. Inet6AddressImpl and Inet4AddressImpl are symetric classes implementing InetAddressImpl. Both will make native call to the "net" library - so both require the library to be loaded.

In the JDK, these two classes are only loaded by InetAddress, after the "net" library has been loaded.

The test test java/net/InetAddress/NullCharInHostnameDriver.java seems to be an outlier (@AlekseiEfimov ?) which for some reason uses a private API (the test is injected in java.base) an create a new instance of Inet6AddressImpl before InetAddress is loaded. This is why without this change we get the UnsatisfiedLinkError. We would have gotten the same from Inet4AddressImpl if the test had first required access to Inet4AddressImpl instead.

So it seems that we should either add the call to load the "net" library to both Inet6AddressImpl and Inet4AddressImpl for symetry (there doesn't seem to be any reason why Inet6AddressImpl would be preferred in this respect),  or remove this call from Inet6AddressImpl and add it to the test instead. Adding the call to jdk.internal.loader.BootLoader.loadLibrary("net"); in the test should work since the actual test class (NullCharInHostname.java) is injected in java.base.

Reply:
Thanks. I'll try adding it to the test.

Reply:
In f57b6f1 I reverted the Inet6AddressImpl change and added the "net" library load to NullCharInHostname.java. With this change the test passed on Unix.